pyface/ui/wx/font_dialog.py

```diff
@@ -0,0 +1,58 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+""" A dialog that allows the user to select a font. """
+
+import wx
+
+from traits.api import provides
+
+from pyface.font import Font, PyfaceFont
+from pyface.i_font_dialog import IFontDialog
+from .dialog import Dialog
+
+
+@provides(IFontDialog)
+class FontDialog(Dialog):
+    """ A dialog for selecting fonts.
+    """
+
+    #: A Font instance that holds the initial font at the start and tracks
+    #: the selected font during user interactions.
+    font = PyfaceFont()
+
+    # ------------------------------------------------------------------------
+    # 'IDialog' interface.
+    # ------------------------------------------------------------------------
+
+    def _create_contents(self, parent):
+        # In wx this is a canned dialog.
+        pass
+
+    # ------------------------------------------------------------------------
+    # 'IWindow' interface.
+    # ------------------------------------------------------------------------
+
+    def close(self):
+        font_data = self.control.GetFontData()
+        wx_font = font_data.GetChosenFont()
+        self.font = Font.from_toolkit(wx_font)
```

Comment:
Clicking `Cancel` on the dialog raises this error:
```
>>> get_font(None, "Helvetica")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/icernyte/Documents/repos/pyface/pyface/font_dialog.py", line 36, in get_font
    result = dialog.open()
  File "/Users/icernyte/Documents/repos/pyface/pyface/i_dialog.py", line 168, in open
    self.close()
  File "/Users/icernyte/Documents/repos/pyface/pyface/ui/wx/font_dialog.py", line 46, in close
    self.font = Font.from_toolkit(wx_font)
  File "/Users/icernyte/Documents/repos/pyface/pyface/font.py", line 387, in from_toolkit
    return cls(**toolkit_font_to_properties(toolkit_font))
  File "/Users/icernyte/Documents/repos/pyface/pyface/ui/wx/font.py", line 162, in toolkit_font_to_properties
    family = wx_family_to_generic_family[toolkit_font.GetFamily()]
wx._core.wxAssertionError: C++ assertion "IsOk()" failed at /Users/robind/projects/bb2/dist-osx-py36/build/ext/wxW
idgets/src/common/fontcmn.cpp(407) in GetFamily(): invalid font
```

Reply:
I'll have a play with this - I think it might require exposing (at least at the private level) methods for getting and setting the font, but I'm OK with that as they can then be used to push common code into the mixin.